controlplane/kubeadm/internal/controllers/controller.go

```diff
@@ -535,6 +571,72 @@ func (r *KubeadmControlPlaneReconciler) ClusterToKubeadmControlPlane(o client.Ob
 	return nil
 }
 
+func (r *KubeadmControlPlaneReconciler) adjustManagedFields(ctx context.Context, m *clusterv1.Machine) error {
```

Comment:
This will be dropped in favor of the ssa util package that will be introduced in https://github.com/kubernetes-sigs/cluster-api/pull/7921.